Added test case for a future exactness ranking score

Toflar opened the pull request.

Toflar commented:
Here we go, this was a lot easier with the new abstraction now ;) Happy for reviews @daun

daun commented:
Looks great 🎉 Nothing much to add besides that. The new rankings all seem reasonable!

Toflar commented:
Noticed that `exactness` is basically the same as 0 typos. So while on it I also implemented the `typo` ranker. Also moved the ranker validation to the `Configuration` itself. This looks good now to me 😊
